docs/operations/security-overview.md

```diff
@@ -264,3 +318,29 @@ As an alternative to using the basic metadata authenticator, as shown in the pre
 
 
 Congratulations, you have configured permissions for user-assigned roles in Druid!
+
+
+## Druid security trust model
+Like all other security systems, trust is the foundation of the Druid security model. Druid administrators and read-only users are trusted users. Therefore, they are not expected to act maliciously.
+
+
+Based on this expectation, Druid operates according to the following principles:
+
+From the inner most layer:
+1. Druid processes run within the system user context. They have access to the local files granted to the specified system user.
+2. The Druid ingestion system can create new processes to execute tasks. Those tasks inherit the user of their parent process. This means that any user authorized to submit an ingestion task can use the ingestion task permissions to read or write any local files that the Druid process has access to.
+
+> Note: Only grant the permission to submit ingestion tasks to trusted users because they can read and write to local file system.
+
+Within the cluster:
+1. Druid assumes it operates on an isolated, protected network where no reachable IP within the network is under adversary control. It is the responsibility of system implementers to setup firewalls and other methods of protection.
```

Comment:
This is a little vague. Is the operator supposed to protect incoming connections to Druid or outgoing or both?

As it's written it seems like the expectation is that restrictions should be set up for both incoming and outgoing connections. Most systems I know of primarily concern themselves with incoming connections, but I'm not a security expert.

Reply:
I think you may need to secure incoming and outgoing, no? If Druid runs as a user that could have permissions may have access to other network resources. This almost begs the question of do we need a security whitepaper type doc with illustrative examples of the best practices. (Not the goal of this PR).

Here's my attempt to clarify FWIW:
> Druid assumes it operates on an isolated, protected network where no reachable IP within the network is under adversary control. When you implement Druid, take care to setup firewalls and other security measures to secure both inbound and outbound connections.